lib/puppet/type/iis_site.rb

```diff
@@ -0,0 +1,270 @@
+require 'puppet/parameter/boolean'
+
+Puppet::Type.newtype(:iis_site) do
+  @doc = "Create a new IIS website."
+
+  newproperty(:ensure) do
+    desc "Specifies whether a site should be present or absent. If present is
+      specified, the site will be created but left in the default stopped state.
+      If started is specified, then the site will be created as well as started.
+      If stopped is specified, then the site will be created and kept stopped."
```

Comment:
I'm not sure? another option would be to forgo `present|absent` and have only `started|stopped` - I believe this is what the `service` type does. e.g., if `ensure => stopped` and site doesn't exist, the equivalent of `present` is assumed.